### ISSUE TITLE
A friendlier LBFGS constructor using a reference

### ISSUE THREAD

**Issue by lict99** [KGrewal1/candle-optimisers#26]
https://github.com/KGrewal1/candle-optimisers/blob/d9533bcd48f40c02a4c10221e4b2e7c0344e7793/src/lbfgs.rs#L137

Hi candle-optimisers maintainer(s),

I'd like to start by saying thank you for this excellent library.

I've been using LBFGS and had a small suggestion. I noticed that `LBFGS::new()` takes ownership of the `model` . This makes it a little tricky to use the data contained in `model` after optimization, since they are "stuck" inside the optimizer. I understand that I can clone the `model`, but sometimes it is expensive.

I believe the optimizer only needs to borrow the `model` to call `loss()` and update the gradients. Would it be possible for `LBFGS::new()` to accept a reference (`&model`) instead? That way, the user could keep ownership of their model and use them easily after the training loop finishes.

Thank you for considering this idea and for the great work on the crate!

**Comment by KGrewal1** [comment]
Yes agree this is a good idea - only one downside is versioning - wanted to keep this in step with candle versions - think best may be to have an alpha release along with this so can stick to this versioning schema
